cogs/base.py

```diff
@@ -0,0 +1,8 @@
+from discord.ext import commands
+
+
+class BaseCog(commands.Cog):
+
+    def __init__(self, bot) -> None:
```

Comment:
Okay, after deliberation and digging through the discord.py codebase, we found that `commands.Cog` doesn't have a default `__init__` function and doesn't store `bot` for us on `self` automatically. https://github.com/Rapptz/discord.py/blob/master/discord/ext/commands/cog.py

Since we want `self.bot` to be available for every Cog in case we add tasks to any Cog (tasks need some kind of reference to bot to be able to interact with the bot since they don't have any `ctx` to use), lets stick with this Base bot as you have done.